core/base/src/main/java/org/eclipse/dataspaceconnector/core/event/EventExecutorServiceContainer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Masatake Iwasaki (NTT DATA) - initial code
+ *
+ */
+
+package org.eclipse.dataspaceconnector.core.event;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Holder class for a shared {@link ExecutorService} across event router implementations.
+ */
+public class EventExecutorServiceContainer {
```

Comment:
Why is this necessary? Why not register the `ExecutorService` directly?

Reply:
@paullatzelsperger I was wondering the same, then I saw that this pattern was already used by `DataTransferExecutorServiceContainer` in the `data-plane-spi` module.
Registering the `ExecutorService` directly would mean that there'll be only one `ExecutorService` registered and used by any extension, while I would prefer every context/extension to have the possibility to have their executor.
For this I would provide a standardize way to configure executors that would be used by different extensions, like keeping a `ExecutorServiceContexts` (or similar) that could be configured by every extension, in a way similar to the one we used in for the api subsystem.
Will open an issue with a proposal.

Reply:
@paullatzelsperger I followed the existing convention for registering distinctive type for ExecutorService used by EventRouter as @ndr-brt mentioned.

Reply:
@ndr-brt @iwasakims makes sense to me.